File: frontend/src/utils/document-status.ts
import { DOCUMENT_STATUS, DOCUMENT_STATUS_CODE } from '@/constants';

export type DocumentStatusCode = typeof DOCUMENT_STATUS_CODE[keyof typeof DOCUMENT_STATUS_CODE];
export type DocumentStatusKey = typeof DOCUMENT_STATUS[keyof typeof DOCUMENT_STATUS];

/**
 * 将后端数值状态码映射为前端文档状态键。
 */
export function mapDocumentStatusFromCode(
  status: number | string | null | undefined,
): DocumentStatusKey {
  const code = typeof status === 'string' ? Number(status) : status;
  switch (code) {
    case DOCUMENT_STATUS_CODE.PUBLISHED:
      return DOCUMENT_STATUS.PUBLISHED;
    case DOCUMENT_STATUS_CODE.ARCHIVED:
      return DOCUMENT_STATUS.ARCHIVED;
    case DOCUMENT_STATUS_CODE.PENDING_REVIEW:
      return DOCUMENT_STATUS.PENDING_REVIEW;
    case DOCUMENT_STATUS_CODE.DRAFT:
    default:
      return DOCUMENT_STATUS.DRAFT;
  }
}

/**
 * 将前端状态键或数值统一解析为数值状态码。
 */
export function resolveDocumentStatusCode(
  status: number | string | null | undefined,
): DocumentStatusCode {
  if (typeof status === 'number') {
    return status as DocumentStatusCode;
  }
  if (status === DOCUMENT_STATUS.PUBLISHED || status === 'published') {
    return DOCUMENT_STATUS_CODE.PUBLISHED;
  }
  if (status === DOCUMENT_STATUS.ARCHIVED || status === 'archived') {
    return DOCUMENT_STATUS_CODE.ARCHIVED;
  }
  if (status === DOCUMENT_STATUS.PENDING_REVIEW || status === 'pending_review') {
    return DOCUMENT_STATUS_CODE.PENDING_REVIEW;
  }
  return DOCUMENT_STATUS_CODE.DRAFT;
}

File: backend/kb-common/src/main/java/com/knowledge/base/common/enums/DocumentStatus.java
package com.knowledge.base.common.enums;

import lombok.Getter;

/**
 * 文档状态枚举
 *
 * <p>与 kb_foundation.kb_dict（dict_code=document_status）及 kb_document.status 字段对齐：</p>
 * <ul>
 *   <li>0 — 草稿（DRAFT）</li>
 *   <li>1 — 已发布（PUBLISHED）</li>
 *   <li>2 — 已归档（ARCHIVED）</li>
 *   <li>3 — 待审核（PENDING_REVIEW）</li>
 * </ul>
 * <p>审核驳回不单独占文档状态，业务回退为 DRAFT(0)。</p>
 *
 * @author 苏三
 * @since 1.0.0
 */
@Getter
public enum DocumentStatus {

    /**
     * 草稿
     */
    DRAFT(0, "草稿"),

    /**
     * 已发布
     */
    PUBLISHED(1, "已发布"),

    /**
     * 已归档
     */
    ARCHIVED(2, "已归档"),

    /**
     * 待审核
     */
    PENDING_REVIEW(3, "待审核");

    private final Integer code;
    private final String name;

    DocumentStatus(Integer code, String name) {
        this.code = code;
        this.name = name;
    }

    /**
     * 获取ByCode。
     */
    public static DocumentStatus getByCode(Integer code) {
        for (DocumentStatus status : values()) {
            if (status.getCode().equals(code)) {
                return status;
            }
        }
        return null;
    }
}
